packages/venia-ui/lib/RootComponents/Search/search.js

```diff
@@ -129,17 +66,67 @@ const Search = props => {
         } else {
             shouldClear.current = true;
         }
-    }, [clearFilters, queryValue]);
+    }, [clearFilters, urlQueryValue]);
 
-    // redirect to the home page if the query doesn't contain input
-    if (!queryValue) {
+    // Redirect to the home page if the query doesn't contain input.
+    if (!urlQueryValue) {
         return <Redirect to="/" />;
     }
 
+    const apolloQueryVariable = categoryId
+        ? { inputText: urlQueryValue, urlCategoryId }
+        : { inputText: urlQueryValue };
+
+    // We don't have to worry about having the same number of hooks because we've
+    // abandoned the render at this point.
```

Comment:
Actually, I'm pretty sure that's an [antipattern](https://stackoverflow.com/questions/54938236/can-you-early-return-with-react-hooks). Even though the early return in this case is a redirect (so we probably won't end up re-rendering this component), I'm not comfortable with this.

Reply:
Thanks for commenting on this, I meant to point this out. I'll see what I can do here.

Reply:
Okay, I updated this (in https://github.com/magento/pwa-studio/pull/1827/commits/c551ef3c3b92bb7aeda59f748ac02ba05e5a18a8).

If we don't have a `query` URL param we used to bail and redirect to the homepage.
Now, we issue the search GraphQL query with an empty string `inputText`. That will result in an `error`.

I updated the rendered output in the `error` case to reflect that the search term may be missing or invalid.

There is no more redirect. I could have added one after a few seconds delay or something but I thought that would be a bit messy.

Let me know what you think, open for opinions on this one.

cc @soumya-ashok @sirugh

Reply:
I think that's a good choice. We can also carve out a case for when we know there's no search term, and show some friendly message prompting the user to enter a search term (rather than one indicating there's been an error).